### DISCUSSION TITLE
Update data based on OL

### DISCUSSION THREAD

**Discussion by rYR79435** [mateusz-bak/openreads#697] [Ideas]
**Is your feature request related to a problem? Please describe.**
I have added books in OpenReads that didn't have much info on OpenLibrary. It would be nice if OpenReads could re-fetch and update certain/all fields.

**Describe the solution you'd like**
And additional button in the book view (from where you can delete/edit/add to favourites) that would trigger the re-fetch.

**Describe alternatives you've considered**
Currently I just add the same book - with the "fresh" OpenLibrary data - again and then delete the old one. This is a pain because I have to transfer the finished reading date.

**Comment by mateusz-bak** [comment]
Hi @rYR79435 thanks for the idea. It would be nice to refetch the data if new is available based on ISBN or Open Library ID. I will check how this can be introduced.

**Comment by yannjor** [comment]
Related to this, It would be nice if book cover data could also be fetched from Open Library. When I imported my books from Goodreads, all my books were without cover, so I had to manually re-add them all from OL to get the cover :D

**Comment by mateusz-bak** [comment]
> Related to this, It would be nice if book cover data could also be fetched from Open Library. When I imported my books from Goodreads, all my books were without cover, so I had to manually re-add them all from OL to get the cover :D

Already implemented, waiting for release in 2.4.0. You can download covers for all books that have ISBN set up
